pallets/pallet-bank/src/lib.rs

```diff
@@ -328,4 +548,33 @@ impl<T: Config> Pallet<T> {
             });
         }
     }
+
+    /// Get the lock id to store into the LockedFund.
+    fn next_lock_id() -> LockId {
+        NextLockId::<T>::mutate(|id| {
+            *id += 1u64;
+            *id
+        })
+    }
+
+    ///Transfer locked funds to free funds
+    fn unlock(account_id: &T::AccountId, locked_id: LockId) {
+        Accounts::<T>::mutate(account_id, |account_data| {
+            if let Some(index) = account_data
```

Comment:
I think part of the logic is similar to how auditor unlocks funds. Maybe you can refactor these 2 function with shared code.